src/main/java/it/smartcommunitylab/aac/webauthn/service/WebAuthnRpService.java

```diff
@@ -210,23 +222,32 @@ private String generateNewKey() {
                     }
                 }
                 if (toUpdate.isEmpty()) {
-                    return Optional.empty();
+                    throw new WebAuthnAuthenticationException(account.getSubject(),
+                            "Could not find the requested credential in the account");
                 }
                 WebAuthnCredential credential = toUpdate.get();
                 credentials.remove(credential);
                 credential.setSignatureCount(result.getSignatureCount());
                 credential.setLastUsedOn(new Date());
                 webAuthnCredentialsRepository.save(credential);
-                return Optional.of(account.getUsername());
+                final String username = account.getUsername();
+                if (username != null) {
+                    return username;
+                } else {
+                    throw new WebAuthnAuthenticationException(account.getSubject(),
+                            "Could not find the requested credential in the account");
+                }
+
             }
         } catch (Exception e) {
         }
-        return Optional.empty();
+        throw new WebAuthnAuthenticationException("_",
```

Comment:
as before, put the exception throw in the catch block

Reply:
Done with 5d0907ea035c889865807dd02ec68ab82544ca9d